libs/frontend/ui-build/src/components/BuildDndOverlay.vue

```diff
@@ -79,35 +77,157 @@ const overlayStyle = computed(() => {
+
+    if (buildContentWindowInner && componentElement) {
+      const { top: buildContentWindowTop, left: buildContentWindowLeft } =
+        buildContentWindowInner.getBoundingClientRect()
+
+      const {
+        top: elementTop,
+        left: elementLeft,
+        width: elementWidth,
+        height: elementHeight,
+      } = componentElement.getBoundingClientRect()
+
+      const elementRelativeTop = elementTop - buildContentWindowTop
+      const elementRelativeLeft = elementLeft - buildContentWindowLeft
+
+      const { top, left, width, height } = normalizeDimensions(
+        elementRelativeTop,
+        elementRelativeLeft,
+        elementWidth,
+        elementHeight,
+        buildContentWindowInner,
+      )
+
+      selectionOverlayStyle.value = {
+        top: `${top}px`,
+        left: `${left}px`,
+        width: `${width}px`,
+        height: `${height}px`,
+        border: '1.5px solid #3768FF',
+      }
+    }
+  } else {
+    selectionOverlayStyle.value = undefined
+  }
+}
+
+watch(
+  () => ({
+    selectedComponent: editor.value?.selectedComponent,
+    // Re-compute selection overlay width & height when active breakpoint changes.
+    activeBreakpoint: activeBreakpoint.value,
+  }),
+  async ({ selectedComponent }) => {
+    computeSelectionOverlay(selectedComponent)
+  },
+  {
+    deep: true,
+  },
+)
+
+const computeRelativeRect = (
+  buildContentWindowInner: HTMLElement,
+  element: HTMLElement,
+) => {
+  const { top: buildContentWindowTop, left: buildContentWindowLeft } =
+    buildContentWindowInner.getBoundingClientRect()
+
+  const {
+    top: elementTop,
+    right: elementRight,
+    bottom: elementBottom,
+    left: elementLeft,
+    width: elementWidth,
+    height: elementHeight,
+  } = element.getBoundingClientRect()
+
+  const relativeTop = elementTop - buildContentWindowTop
+  const relativeRight = elementRight - buildContentWindowLeft
+  const relativeBottom = elementBottom - buildContentWindowTop
+  const relativeLeft = elementLeft - buildContentWindowLeft
+
+  return {
+    relativeTop,
+    relativeRight,
+    relativeBottom,
+    relativeLeft,
+    elementWidth,
+    elementHeight,
+  }
+}
+
+// Divide offsets by `builderScale` because `.build-content-window-inner` scales
+// based on `builderScale` in the editor context.
+const normalizeDimensions = (
+  overlayTop: number,
+  overlayLeft: number,
+  overlayWidth: number,
+  overlayHeight: number,
+  buildContentWindowInner: HTMLElement,
+) => {
+  const builderScale = editor.value?.builderScale ?? 1
+
+  let top = overlayTop / builderScale
+  let left = overlayLeft / builderScale
+  const width = overlayWidth / builderScale
+  const height = overlayHeight / builderScale
+
+  top += buildContentWindowInner.scrollTop
+  left += buildContentWindowInner.scrollLeft
+
+  return {
+    left,
+    top,
+    width,
+    height,
+  }
+}
+
+onMounted(() => {
+  setTimeout(() => computeSelectionOverlay(editor.value?.selectedComponent), 1)
```

Comment:
I tried to solve the selected component border on load bug with this. It sort of works, but doesn't quite draw the border in the right place.

This is what the border normally looks like:
![Screenshot 2024-01-10 at 12 16 07 AM](https://github.com/samatechtw/pubstudio-builder/assets/550982/344b3b93-3861-47a6-a92f-5786a5e1b6c6)

And then on load (after the fix):

![Screenshot 2024-01-10 at 12 35 31 AM](https://github.com/samatechtw/pubstudio-builder/assets/550982/4070adaa-f321-4659-8025-dac46b41233d)

I don't quite understand the code yet, so I'm not sure why it would be slightly more or less offset sometimes. I noticed that clicking to/from a component sometimes causes the offset to be different as well.

Reply:
Got it, I'll look deeper into this and try to fix it today.
Thank you so much for helping me out!

Reply:
Hi @sampullman 

Unfortunately, I still didn't manage to solve this problem 😕.
But I noticed that the selection overlay will cause the root component to shrink a little bit when selected:

https://github.com/samatechtw/pubstudio-builder/assets/59676941/8c2da393-1409-4198-9192-2920e2ac6d55

Notice how the build content "grow" to the right when root component unselected, and shrink a little bit to the left when re-selected. Removing the selection overlay seems to fix this glitch; maybe this has something to do with the bug we're seeing, but I'm not 100% sure.

I think I'll go pick up some other issues so that I don't get stuck here for too long. Sorry about that! 😵

Reply:
Ok, I'll try to fix it when I get the chance. Meanwhile I think I'll cherry pick the other commits and merge them first!